netadapter/server/grpcserver/grpc_connection.go

```diff
@@ -0,0 +1,93 @@
+package grpcserver
+
+import (
+	"net"
+	"sync/atomic"
+
+	"github.com/pkg/errors"
+
+	"github.com/kaspanet/kaspad/netadapter/server"
+	"github.com/kaspanet/kaspad/netadapter/server/grpcserver/protowire"
+	"github.com/kaspanet/kaspad/wire"
+	"google.golang.org/grpc"
+)
+
+type gRPCConnection struct {
+	server                *gRPCServer
+	address               net.Addr
+	sendChan              chan *protowire.KaspadMessage
+	receiveChan           chan *protowire.KaspadMessage
+	errChan               chan error
+	clientConn            grpc.ClientConn
+	onDisconnectedHandler server.OnDisconnectedHandler
+
+	isConnected int32
+}
+
+func newConnection(server *gRPCServer, address net.Addr) *gRPCConnection {
+	connection := &gRPCConnection{
+		server:      server,
+		address:     address,
+		sendChan:    make(chan *protowire.KaspadMessage),
+		receiveChan: make(chan *protowire.KaspadMessage),
+		errChan:     make(chan error),
+		isConnected: 1,
+	}
+
+	server.addConnection(connection)
+
+	return connection
+}
+
+func (c *gRPCConnection) IsConnected() bool {
+	return atomic.LoadInt32(&c.isConnected) == 1
+}
+
+func (c *gRPCConnection) SetOnDisconnectedHandler(onDisconnectedHandler server.OnDisconnectedHandler) {
+	c.onDisconnectedHandler = onDisconnectedHandler
+}
+
+// Send sends the given message through the connection
+// This is part of the Connection interface
+func (c *gRPCConnection) Send(message wire.Message) error {
+	messageProto, err := protowire.FromWireMessage(message)
+	if err != nil {
+		return err
+	}
+
+	c.sendChan <- messageProto
+
+	return <-c.errChan
+}
+
+// Receive receives the next message from the connection
+// This is part of the Connection interface
+func (c *gRPCConnection) Receive() (wire.Message, error) {
+	protoMessage := <-c.receiveChan
+	if protoMessage == nil {
+		return nil, errors.New("Connection closed during receive")
+	}
+
+	return protoMessage.ToWireMessage()
+}
+
+// Disconnect disconnects the connection
+// This is part of the Connection interface
+func (c *gRPCConnection) Disconnect() error {
```

Comment:
Please add to the comment that calling this function a second time does nothing.

Reply:
"This is part of such and such interface" usually comes at the end of the comment. Please reshuffle the lines to make it so.